tests/unit/CMakeLists.txt

```diff
@@ -0,0 +1,125 @@
+# @file      CMakeLists.txt
+# @brief     Unit Tests for Mender MCU platform independent code
+#
+# Copyright Northern.tech AS
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.16.3)
+project(mender-mcu-tests-unit)
+
+# TODO: move somewhere shared with smoke tests and remove the other fetch method
+include(FetchContent)
+FetchContent_Declare(
+  cjson
+  GIT_REPOSITORY https://github.com/DaveGamble/cJSON
+  GIT_TAG        v1.7.15
+  DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+)
+
+set(ENABLE_CJSON_TEST OFF CACHE INTERNAL "Turn off tests")
+FetchContent_MakeAvailable(cjson)
+
+# TODO: Investigate a better way to do this. Maybe a more recent version of cJSON better
+# supports build as external library?
+include_directories(${CMAKE_CURRENT_BINARY_DIR}/_deps/cjson-src/)
+
+# TODO: Add mbedTLS dependency
+
+# Include mender-mcu-client library
+add_subdirectory("${CMAKE_CURRENT_LIST_DIR}/../../target/posix" build)
+
+enable_testing()
+
+option(COVERAGE "Turn coverage instrumentation on (Default: OFF)" OFF)
+if($CACHE{COVERAGE})
+  set(CMAKE_CXX_FLAGS "--coverage $CACHE{CMAKE_CXX_FLAGS}")
+endif()
+
+set(GTEST_VERSION 1.12.1)
+
+option(MENDER_DOWNLOAD_GTEST "Download google test if it is not found (Default: ON)" ON)
+
+if (MENDER_DOWNLOAD_GTEST)
+
+  ### BEGIN taken from https://google.github.io/googletest/quickstart-cmake.html
+  include(FetchContent)
+  FetchContent_Declare(
+    googletest
+    URL https://github.com/google/googletest/archive/refs/tags/release-${GTEST_VERSION}.zip
+    DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+  )
+
+  # For Windows: Prevent overriding the parent project's compiler/linker settings
+  set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+  ### END
+
+  set(BUILD_GMOCK ON)
+  set(INSTALL_GTEST OFF)
+  FetchContent_MakeAvailable(googletest)
+
+else()
+  find_package(GTest REQUIRED)
+endif()
+
+if($CACHE{COVERAGE})
+  add_custom_target(coverage_enabled COMMAND true)
+else()
+  add_custom_target(coverage_enabled
+    COMMAND echo 'Please run `cmake -D COVERAGE=ON .` first!'
+    COMMAND false
+  )
+endif()
+
+add_custom_target(coverage
+  COMMAND lcov --capture --quiet --directory .
+               --output-file coverage.lcov
+               --exclude '/usr/*'
+               --exclude '*/googletest/*'
+               --exclude '*_test.*'
+#               --exclude '*/googlemock/*'
```

Comment:
Why is this commented out?

Reply:
We will need when adding tests that make user of it, for example with `EXPECT_THAT` type of macros. But I think I can leave it un-commented, actually - it shouldn't matter that the pattern does not match.

Reply:
Actually not :laughing: 

```
geninfo: ERROR: 'exclude' pattern '*/googlemock/*' is unused.
```

I will leave it around if you don't mind. It will come in handy when writing tests, I am pretty sure.